interface.go

```diff
@@ -0,0 +1,55 @@
+package event
+
+import (
+	"errors"
+	"reflect"
+)
+
+type SubSettings struct {
+	forcedType reflect.Type
+}
+type SubOption func(*SubSettings) error
+
+func ForceSubType(evtType interface{}) SubOption {
+	return func(s *SubSettings) error {
+		typ := reflect.TypeOf(evtType)
+		if typ.Kind() != reflect.Ptr {
+			return errors.New("ForceSubType called with non-pointer type")
+		}
+		s.forcedType = typ
+		return nil
+	}
+}
+
+type EmitterSettings struct{
+	makeStateful bool
+}
+type EmitterOption func(*EmitterSettings)
+
+func Stateful(s *EmitterSettings) {
+	s.makeStateful = true
+}
+
+type Bus interface {
+	// Subscribe creates new subscription. Failing to drain the channel will cause
+	// publishers to get blocked
+	Subscribe(typedChan interface{}, opts ...SubOption) (CancelFunc, error)
```

Comment:
Can we make sure to guarantee (and document) that `CancelFunc` returns _after_ the last send on the channel?

Reply:
CancelFunc here takes a write lock on node, so that's only the question of docs.